core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1350,12 +1350,14 @@ wasm_runtime_load_ex(uint8 *buf, uint32 size, const LoadArgs *args,
                                           true,
 #endif
                                           args, error_buf, error_buf_size);
+        ((WASMModule *)module_common)->is_binary_freeable = true;
 #endif
     }
     else if (get_package_type(buf, size) == Wasm_Module_AoT) {
 #if WASM_ENABLE_AOT != 0
         module_common = (WASMModuleCommon *)aot_load_from_aot_file(
             buf, size, args, error_buf, error_buf_size);
+        ((AOTModule *)module_common)->is_binary_freeable = true;
```

Comment:
Should be `((AOTModule *)module_common)->is_binary_freeable = args.wasm_binary_freeable;`